File: treino-security-noJWT/src/main/java/com/treino_security_noJWT/controllers/UserController.java
package com.treino_security_noJWT.controllers;

import java.util.List;

import org.springframework.http.ResponseEntity;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.treino_security_noJWT.Models.User;
import com.treino_security_noJWT.Models.UserDTO;
import com.treino_security_noJWT.services.UserService;

import jakarta.validation.Valid;

@RestController
@RequestMapping("/users")
public class UserController {

    private UserService userService;
    private PasswordEncoder passwordEncoder;

    public UserController(UserService userService, PasswordEncoder passwordEncoder) {
        this.userService = userService;
        this.passwordEncoder = passwordEncoder;
    }

    @GetMapping
    public ResponseEntity<String> teste(){
        return ResponseEntity.ok().body("testeeeee");
    }

    @PostMapping
    public ResponseEntity<String> CreateUser(@RequestBody @Valid UserDTO user){
        User newUser = new User();
        newUser.setLogin(user.login());
        newUser.setPassword(passwordEncoder.encode(user.senha()));
        userService.createUser(newUser);
        return ResponseEntity.ok().body("Usuário Cadastrado");
    }

    @GetMapping("/allUsers")
    public ResponseEntity<List<User>> findAll(){
        List<User> allUser = userService.listAll();
        return ResponseEntity.ok().body(allUser);
    }

}

File: track-found-TS/track-found-backend/src/controllers/UserController.ts
import { User, UserDTO } from "../models/User";
import * as userService from "../services/UserService";
import { Request, Response } from "express";

export const createUser = async(req:Request, res:Response) => {
    try{
        const {name, email,password} = req.body;
        const user:User = {name:name, email:email,password:password}
        const newUser:UserDTO = await userService.createUser(user)
        res.status(201).json(newUser)
    }catch(error){
        if(error instanceof Error){
            return res.status(400).json({ error: error.message })
        }
        return res.status(500).json({ error: "Erro interno no servidor ao criar usuário" });
    }
}